### ISSUE TITLE
Consider the addition of the pre-trained LASSO model engine

### ISSUE THREAD

**Issue by mattwarkentin** [tidymodels/parsnip#1146]
Hi `tidymodels` team,

I am not sure if you have seen the recent release of the "pre-trained LASSO" which is now available in the package `ptLasso`. Not sure if it will find a home here in the `tidymodels` ecosystem, but I just wanted to bring it to your attention and open some discussion about its potential implementation.

The sales-pitch (from them):
> __Can pretraining help the lasso? Yes!__
>
> Here we present a framework for the lasso in which an overall model is fit to a large set of data, and then fine-tuned to a specific task. This latter dataset can be a subset of the original dataset, but does not need to be. Pretraining the lasso has a wide variety of applications, including stratified models, multinomial targets, multi-response models, conditional average treatment estimation and even gradient boosting.

Paper: http://arxiv.org/pdf/2401.12911
Website: https://erincr.github.io/ptLasso/
R package: https://erincr.github.io/ptLasso/
Other helpful docs: https://www.erincraig.me/lasso-pretraining
